pkg/apis/serving/v1alpha2/kfservice_types.go

```diff
@@ -21,36 +21,65 @@ import (
 
 // KFServiceSpec defines the desired state of KFService
 type KFServiceSpec struct {
-	Default ModelSpec `json:"default"`
+	Default ComponentsSpec `json:"default"`
 	// Canary defines an alternate configuration to route a percentage of traffic.
-	Canary               *ModelSpec `json:"canary,omitempty"`
-	CanaryTrafficPercent int        `json:"canaryTrafficPercent,omitempty"`
+	Canary               *ComponentsSpec `json:"canary,omitempty"`
+	CanaryTrafficPercent int             `json:"canaryTrafficPercent,omitempty"`
 }
 
-// ModelSpec defines the configuration to route traffic to a predictor.
-type ModelSpec struct {
+type ComponentsSpec struct {
+	// predict defines the model serving spec
+	// +required
+	Predict ModelSpec `json:"predict"`
```

Comment:
While we're changing things, should we update `ModelSpec` to be `PredictorSpec`?

Reply:
or `PredictSpec`

Reply:
+1 to PredictSpec.

Reply:
Makes sense to understand conceptually, predict/explain/transform are verbs, and define actions/functions, and the spec is the input configuration. Only concern which comes later is Tensorflow Spec, XGBoost Spec etc. are subsets of ModelSpec currently, and not sure if they conceptually fit nicely under PredictSpec......?

Reply:
Then does it make sense to have everything as a 'noun'?  Predictor, Explainer, Transformer, Component (Endpoint).....?

Reply:
If everyone agrees I am ok to change to noun :)

Reply:
discussed this with @ellis-bigelow early on in the meeting.... think we agreed on using 'noun' throughout

Reply:
+1!